.NET Framework and .NET Core project files conflict with each other

See https://github.com/microsoft/msbuild/issues/1924.

The symptom is that if you build a ***-NetCore.csproj project in a directory and then the older ***.csproj file in the same directory you get an extremely unhelpful error message:

```
a483e79acc1e:Model salkeldr$ dotnet build Model-NetCore.csproj 
Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 2.71 sec for /Users/salkeldr/Documents/GitHub/boogie/Source/CodeContractsExtender/CodeContractsExtender-NetCore.csproj.
  Restore completed in 2.71 sec for /Users/salkeldr/Documents/GitHub/boogie/Source/Model/Model-NetCore.csproj.
  CodeContractsExtender-NetCore -> /Users/salkeldr/Documents/GitHub/boogie/Source/CodeContractsExtender/bin/Debug/netcoreapp3.1/BoogieCodeContractsExtender.dll
  Model-NetCore -> /Users/salkeldr/Documents/GitHub/boogie/Source/Model/bin/Debug/netcoreapp3.1/BoogieModel.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:04.75
a483e79acc1e:Model salkeldr$ msbuild  Model.csproj 
Microsoft (R) Build Engine version 16.3.0-ci for Mono
Copyright (C) Microsoft Corporation. All rights reserved.

Build started 2020-02-05 10:53:00 AM.
Project "/Users/salkeldr/Documents/GitHub/boogie/Source/Model/Model.csproj" on node 1 (default targets).
/Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/xbuild/Microsoft/NuGet/Microsoft.NuGet.targets(186,5): error : Your project does not reference ".NETFramework,Version=v4.5" framework. Add a reference to ".NETFramework,Version=v4.5" in the "TargetFrameworks" property of your project file and then re-run NuGet restore. [/Users/salkeldr/Documents/GitHub/boogie/Source/Model/Model.csproj]
Done Building Project "/Users/salkeldr/Documents/GitHub/boogie/Source/Model/Model.csproj" (default targets) -- FAILED.

Build FAILED.

"/Users/salkeldr/Documents/GitHub/boogie/Source/Model/Model.csproj" (default target) (1) ->
(ResolveNuGetPackageAssets target) -> 
  /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/xbuild/Microsoft/NuGet/Microsoft.NuGet.targets(186,5): error : Your project does not reference ".NETFramework,Version=v4.5" framework. Add a reference to ".NETFramework,Version=v4.5" in the "TargetFrameworks" property of your project file and then re-run NuGet restore. [/Users/salkeldr/Documents/GitHub/boogie/Source/Model/Model.csproj]

    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:00.51

```

The suggested fix seems to be to add a Directory.Build.props in the root of the repo that relocates the `obj` directory for the new project files (modified to fit the convention in Boogie):

```
<Project>
  <PropertyGroup Condition="$(MSBuildProjectName.Contains('-NetCore'))">
    <BaseIntermediateOutputPath>obj_core\</BaseIntermediateOutputPath>
  </PropertyGroup>
</Project>
```

The above solution does not fully solve the problem. Applying it results in the following error in the .NET Core build.

```
obj/Debug/netcoreapp3.1/.NETCoreApp,Version=v3.1.AssemblyAttributes.cs(4,12): error CS0579: Duplicate 'global::System.Runtime.Versioning.TargetFrameworkAttribute' attribute [/home/bkragl/Code/boogie/Source/CodeContractsExtender/CodeContractsExtender-NetCore.csproj]
```

It can be fixed by adding `<GenerateTargetFrameworkAttribute>false</GenerateTargetFrameworkAttribute>` to the confiuration, but I don't really want to do that because (a) I don't fully understand the consequences of omitting this information form the assemblies, and, more importantly, (b) the error above indicates that there might be other strange interferences between the two builds.

So IMO we should add some troubleshooting information to `README.md` that warns from running both builds in the same directory.

--- a/.travis.yml
+++ b/.travis.yml
@@ -35,7 +35,7 @@ script:
   # Run unit tests
   - dotnet test --no-build -c ${CONFIGURATION} ${SOLUTION}
   # Run lit test suite
-  - lit -v --timeout=120 -D dotnet -D configuration=${CONFIGURATION} Test
+  - lit -v --timeout=120 -D configuration=${CONFIGURATION} Test
   # Do not deploy if GitVersionTask could not be used to extract the correct version number
   - test ! -e Source/BoogieDriver/bin/${CONFIGURATION}/Boogie.1.0.0.nupkg
 deploy:

--- a/README.md
+++ b/README.md
@@ -45,7 +45,7 @@ Boogie releases are packaged as a .NET Core global tool available at
 $ dotnet tool install --global boogie
 ```
 
-To run Boogie, a supported SMT solver has to be available (see below).
+To run Boogie, a supported SMT solver has to be available (see [below](#backend-smt-solver)).
 
 ## Building
 
@@ -53,7 +53,7 @@ The preferred way to build (and run) Boogie today is using .NET Core.
 Historically, Boogie can also be built using the classic .NET Framework (on
 Windows) and Mono (on Linux/OSX), but this might not be supported in the future.
 
-To run Boogie, a supported SMT solver has to be available (see below).
+To run Boogie, a supported SMT solver has to be available (see [below](#backend-smt-solver)).
 
 ### .NET Core
 
@@ -72,6 +72,8 @@ for a local installation.
 
 ### Windows (requires Visual Studio)
 
+:warning: This build is deprecated. Use [.NET Core](#net-core) if possible.
+
 1. Open ``Source\Boogie.sln`` in Visual Studio.
 2. Right click the ``Boogie`` solution in the Solution Explorer and click ``Restore NuGet Packages``.
 3. Click ``Build > Build Solution``.
@@ -79,6 +81,8 @@ for a local installation.
 The compiled Boogie binary is `Binaries\Boogie.exe`.
 
 ### Linux/OSX (requires Mono)
+
+:warning: This build is deprecated. Use [.NET Core](#net-core) if possible.
 
 Either open ``Source\Boogie.sln`` in MonoDevelop and perform the same steps as
 described for Visual Studio above, of compile on the command line as follows.
@@ -97,6 +101,13 @@ $ msbuild Source/Boogie.sln
 
 The compiled Boogie binary is `Binaries/Boogie.exe`, which can be executed with
 `mono` or using the wrapper script `Binaries/boogie`.
+
+### Troubleshooting
+
+The .NET Core and .NET Framework builds conflict with each other, and thus it is
+not possible to run both builds in the same directory. If you want to switch
+between builds (or encounter mysterious erros as described in #190), remove all
+build outputs (e.g., using `git clean -dfx`).
 
 ## Backend SMT Solver
 
@@ -147,8 +158,8 @@ each time commits are pushed to the master branch (including PR merges). By
 default, the created tag increments the patch version number from the previous
 tag. For example, if the last tagged commit were `v2.4.3`, then pushing to
 master would tag the latest commit with `v2.4.4`. If incrementing minor or major
-number is desired instead of patch, simply add `#minor` or `#major` to the first
-line of the commit message. For instance:
+number is desired instead of patch, simply add `#minor` or `#major` anywhere in
+the commit message. For instance:
 
 > Adding the next greatest feature. #minor
 

--- a/Test/lit.site.cfg
+++ b/Test/lit.site.cfg
@@ -82,7 +82,7 @@ framework = lit_config.params.get('framework', 'netcoreapp3.1')
 
 runtime = ''
 
-if 'dotnet' in lit_config.params:
+if not 'legacy' in lit_config.params:
     boogieBinary = 'Source/BoogieDriver/bin/{}/{}/BoogieDriver.dll'.format(configuration, framework)
     runtime = 'dotnet'
 else:

